course_catalog/etl/mitpe.py

```diff
@@ -0,0 +1,212 @@
+
+    Returns:
+          Decimal: Price of course or None
+    """
+    try:
+        price_text = course.find("div", {"class": "views-field-field-course-fee"}).text
+    except AttributeError:
+        return None
+    price_match = re.search(r"([\d,]+)", price_text)
+    if price_match:
+        return Decimal(price_match.group(0).replace(",", ""))
+
+
+def _parse_run_dates(course):
+    """
+    Extracts the start and end dates for each course run
+
+    Args:
+        course(Tag): The BeautifulSoup tag containing course data
+
+    Returns:
+        list of tuple: List of start and end date tuples for each course run
+
+    """
+    date_div = course.find("span", {"class": "date-display-range"})
+    run_dates = []
+    if date_div:
+        dates = parse_dates(date_div.get_text())
+        if dates:
+            run_dates.append(dates)
+    return run_dates
+
+
+def _parse_instructors(course):
+    """
+    Extract instructor names from the course detail page
+
+    Args:
+        details(Tag): BeautifulSoup Tag for course details
+
+    Returns:
+        list of lists: List of first & last names of each instructor
+
+    """
+    return [
+        instructor.get_text().split(" ", 1)
+        for instructor in course.find(
+            "div", {"class": "views-field-field-lead-instructors"}
+        ).findAll("div", {"class": "field-content"})
+    ]
+
+
+def _parse_description(details):
+    """
+    Extract short description from the course detail page
+
+    Args:
+        details(Tag): BeautifulSoup Tag for course details
+
+    Returns:
+        list of str: List of course paragraph strings
+
+    """
+    div = details.find("div", {"class": "course-right"}).find(
+        "div", {"class": "field--type-text-with-summary"}
+    )
+    # pylint: disable=expression-not-assigned
+    [p.extract() for p in div.findAll("p", {"class": "special"})]
+    paragraphs = []
+    for child in div.findAll():
+        if child.name == "p":
+            paragraphs.append(child.text)
+        elif child.name == "h3":
+            break
+
+    return paragraphs
+
+
+@log_exceptions(
+    "Error extracting MIT Professional Education catalog", exc_return_value=[]
+)
+def extract():
+    """Loads the MIT Professional Education catalog data via BeautifulSoup"""
+    if not settings.MITPE_BASE_URL:
+        log.error("MIT Professional base URL not set, skipping ETL")
+        return []
+    courses = []
+    soup = bs(
+        requests.get(urljoin(settings.MITPE_BASE_URL, "/course-catalog")).content,
+        "html.parser",
+    )
+    listings = soup.find("div", {"class": "course-listing"}).findAll(
+        "div", {"class": "views-row"}
+    )
+    for listing in listings:
+        link = listing.find("a")
+        url = urljoin(settings.MITPE_BASE_URL, link.get("href"))
+        details = bs(requests.get(url).content, "html.parser")
+        description = _parse_description(details)
+        short_description = strip_extra_whitespace(description[0])
```

Comment:
Some of the first paragraphs are a bit too general to be a good course description, but I don't have a better solution